src/crimson/osd/recovery_backend.cc

```diff
@@ -225,54 +225,41 @@ RecoveryBackend::scan_for_backfill(
 {
   logger().debug("{} starting from {}", __func__, start);
   auto version_map = seastar::make_lw_shared<std::map<hobject_t, eversion_t>>();
-  return backend->list_objects(start, max).then_interruptible(
-    [this, start, version_map] (auto&& ret) {
-    auto&& [objects, next] = std::move(ret);
-    return seastar::do_with(
-      std::move(objects),
-      [this, version_map](auto &objects) {
-      return interruptor::parallel_for_each(objects,
-	[this, version_map] (const hobject_t& object)
-	-> interruptible_future<> {
-	crimson::osd::ObjectContextRef obc;
-	if (pg.is_primary()) {
```

Comment:
I'm not sure this is relevant to us. In classic, we have this check to not use `object_contexts` on non primaries. Nevertheless, we would still fallback to "loading" this object.